Shared Infra Property & Quantity Sets fixes

AlexBrad1eyCT opened the pull request:
Shared Infra Property set fixes highlighted during deliverable import. These changes are in the SharedInfrastructureElements schema but are under the remit of Road & P&W

Fixes Shared Infra Items of #326

AlexBrad1eyCT committed:
Shared Infra fixes

AlexBrad1eyCT committed:
Merge remote-tracking branch 'upstream/develop' into feature/Shared-Infra-Fixes

AlexBrad1eyCT committed:
Quantity set errors and a incorrect serialisation

AlexBrad1eyCT committed:
Update DocPropertySet.xml

AlexBrad1eyCT commented:
@larswik or @SergejMuhic a once over would be appreciated these are common infra fixes

larswik commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_TransportElementElevator/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_TransportElementElevator" UniqueId="0LWrk0qX8HuO00025QrE$V" ApplicableType="IfcTransportElement/ELEVATOR" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_TransportElementElevator" UniqueId="0LWrk0qX8HuO00025QrE$V" ApplicableType="IfcTransportElement/IfcTransportElementFixedTypeEnum(.ELEVATOR.),IfcTransportElementType/IfcTransportElementFixedTypeEnum(.ELEVATOR.)" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
Just checking: Syntax (select of enum).

AlexBrad1eyCT replied:
it is correct and saved in IfcDoc

larswik commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/PropertySets/Pset_LinearReferencingMethod/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_LinearReferencingMethod" UniqueId="a4db95cf-06af-4562-8d73-bbd65fbcc8da" ApplicableType="IfcAlignment,IfcReferent/(.POSITION.)">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_LinearReferencingMethod" UniqueId="a4db95cf-06af-4562-8d73-bbd65fbcc8da" ApplicableType="IfcAlignment,IfcReferent/POSITION" PropertySetType="PSET_OCCURRENCEDRIVEN">
```
Need to check if this could be on any kind of IfcReferent => Discuss!

AlexBrad1eyCT replied:
@SergejMuhic

SergejMuhic replied:
Any kind of `IfcReferent`.

AlexBrad1eyCT committed:
Merge branch 'develop' into feature/Shared-Infra-Fixes

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_ElementAssemblyCommon/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_ElementAssemblyCommon" UniqueId="0f3eee41-95a9-45ec-be30-2ea1b18f294e">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_ElementAssemblyCommon" UniqueId="0f3eee41-95a9-45ec-be30-2ea1b18f294e" ApplicableType="IfcElementAssembly,IfcElementAssemblyType" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
```suggestion
<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_ElementAssemblyCommon" UniqueId="0f3eee41-95a9-45ec-be30-2ea1b18f294e" ApplicableType="IfcElementAssembly" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_EnvironmentalImpactIndicators/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_EnvironmentalImpactIndicators" UniqueId="1xexg0qU4HuO00025QrE$V" ApplicableType="IfcElement" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_EnvironmentalImpactIndicators" UniqueId="1xexg0qU4HuO00025QrE$V" ApplicableType="IfcElement,IfcElementType" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
IfcElementType can be reverted.

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_EnvironmentalImpactValues/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_EnvironmentalImpactValues" UniqueId="3_B7e0qU4HuO00025QrE$V" ApplicableType="IfcElement" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_EnvironmentalImpactValues" UniqueId="3_B7e0qU4HuO00025QrE$V" ApplicableType="IfcElement,IfcElementType" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
IfcElementType can be reverted.

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_ProcessCapacity/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_ProcessCapacity" UniqueId="95ac3d01-c324-4bba-912c-4185b0bfce5d" ApplicableType="IfcSpatialElement,IfcSpatialElementType,IfcTransportElement,IfcTransportElementType,IfcBuiltSystem,IfcDistributionSystem,IfcZone,IfcDoor" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_ProcessCapacity" UniqueId="95ac3d01-c324-4bba-912c-4185b0bfce5d" ApplicableType="IfcSpatialElement,IfcSpatialElementType,IfcTransportElement,IfcTransportElementType,IfcBuiltSystem,IfcDistributionSystem,IfcZone,IfcDoor,IfcDoorType" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
IfcDoorType can be reverted.

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_SpaceCommon/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_SpaceCommon" UniqueId="2Vni80qWWHuO00025QrE$V" ApplicableType="IfcSpace" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_SpaceCommon" UniqueId="2Vni80qWWHuO00025QrE$V" ApplicableType="IfcSpace,IfcSpaceType" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
IfcSpaceType can be reverted.

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_SpatialZoneCommon/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_SpatialZoneCommon" UniqueId="0fd7d69f-be3e-44d4-be9a-df6368e9cc6b">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_SpatialZoneCommon" UniqueId="0fd7d69f-be3e-44d4-be9a-df6368e9cc6b" ApplicableType="IfcSpatialZone,IfcSpatialZoneType" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
IfcSpatialZoneType can be reverted.

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_TransportElementCommon/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_TransportElementCommon" UniqueId="3X3v_0qX4HuO00025QrE$V" ApplicableType="IfcTransportElement" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_TransportElementCommon" UniqueId="3X3v_0qX4HuO00025QrE$V" ApplicableType="IfcTransportElement,IfcTransportElementType" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
IfcTransporteElementType can be reverted.

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_TransportElementElevator/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_TransportElementElevator" UniqueId="0LWrk0qX8HuO00025QrE$V" ApplicableType="IfcTransportElement/ELEVATOR" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_TransportElementElevator" UniqueId="0LWrk0qX8HuO00025QrE$V" ApplicableType="IfcTransportElement/IfcTransportElementFixedTypeEnum(.ELEVATOR.),IfcTransportElementType/IfcTransportElementFixedTypeEnum(.ELEVATOR.)" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
```suggestion
<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_TransportElementElevator" UniqueId="0LWrk0qX8HuO00025QrE$V" ApplicableType="IfcTransportElement/ELEVATOR" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_Uncertainty/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_Uncertainty" UniqueId="eb7c91fe-a348-4889-bdfa-65cc36c81917" ApplicableType="IfcProduct,IfcTypeProduct">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_Uncertainty" UniqueId="eb7c91fe-a348-4889-bdfa-65cc36c81917" ApplicableType="IfcProduct,IfcTypeProduct" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
```suggestion
<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_Uncertainty" UniqueId="eb7c91fe-a348-4889-bdfa-65cc36c81917" ApplicableType="IfcProduct" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedFacilitiesElements/PropertySets/Pset_MaintenanceStrategy/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_MaintenanceStrategy" UniqueId="f7f14ce7-afdc-4d4f-915a-7083f3c19a95" ApplicableType="IfcElement,IfcAsset,IfcSystem">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_MaintenanceStrategy" UniqueId="f7f14ce7-afdc-4d4f-915a-7083f3c19a95" ApplicableType="IfcElement,IfcAsset,IfcSystem,IfcElementType" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
IfcElementType can be reverted.

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/PropertySets/Pset_LinearReferencingMethod/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_LinearReferencingMethod" UniqueId="a4db95cf-06af-4562-8d73-bbd65fbcc8da" ApplicableType="IfcAlignment,IfcReferent/(.POSITION.)">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_LinearReferencingMethod" UniqueId="a4db95cf-06af-4562-8d73-bbd65fbcc8da" ApplicableType="IfcAlignment,IfcReferent/POSITION" PropertySetType="PSET_OCCURRENCEDRIVEN">
```
```suggestion
<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_LinearReferencingMethod" UniqueId="a4db95cf-06af-4562-8d73-bbd65fbcc8da" ApplicableType="IfcAlignment,IfcReferent" PropertySetType="PSET_OCCURRENCEDRIVEN">
```

SergejMuhic reviewed (commented):
@AlexBrad1eyCT looks like I will have to ask you for another merge here.

AlexBrad1eyCT commented:
@SergejMuhic  see comment in #337

AlexBrad1eyCT committed:
Merge branch 'develop' into feature/Shared-Infra-Fixes

AlexBrad1eyCT committed:
Updating PSet_TransportElementCommon Name

intermediary element IfcTransportationDevice introduced and assigned this PSet but name has not been updated.

AlexBrad1eyCT committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/PropertySets/Pset_LinearReferencingMethod/DocPropertySet.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

AlexBrad1eyCT committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_TransportElementElevator/DocPropertySet.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

AlexBrad1eyCT committed:
Apply suggestions from code review

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

AlexBrad1eyCT committed:
Type Applicability Convention

AlexBrad1eyCT commented:
Type Applicability Applied

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcTransportElement/Documentation.md:
```diff
@@ -17,4 +17,4 @@ Depending on local classification systems transport elements and transportation
 
 { .change-ifc2x4}
-> IFC4 CHANGE&nbsp; The last attributes CapacityByWeight and CapacityByNumber are removed, use Pset_TransportElementCommon instead.
+> IFC4 CHANGE&nbsp; The last attributes CapacityByWeight and CapacityByNumber are removed, use Pset_TransportationDeviceCommon instead.
```
Wondering whether changing a note to IFC4 makes sense. Add a note to IFC4X3?
